src/studio/src/designer/frontend/app-development/features/dataModelling/components/XSDUploader.tsx

```diff
@@ -7,73 +8,97 @@ import { XSDUploadUrl } from 'utils/urlHelper';
 import axios from 'axios';
 
 interface IXSDUploaderProps {
-  language: any,
-  onXSDUploaded: (filename: string) => void,
+  language: unknown;
+  onXSDUploaded: (filename: string) => void;
 }
 
-export default function XSDUploader(props: IXSDUploaderProps) {
+const anchorOrigin: PopoverOrigin = {
+  vertical: 'bottom',
+  horizontal: 'left',
+};
+
+export default function XSDUploader({
+  onXSDUploaded,
+  language,
+}: IXSDUploaderProps) {
   const [uploadButtonAnchor, setUploadButtonAnchor] = React.useState(null);
   const [uploading, setUploading] = React.useState(false);
   const [errorText, setErrorText] = React.useState(null);
-  const onUploadClick = (event: any) => {
+
+  const handleUploadClick = React.useCallback((event) => {
     setUploadButtonAnchor(event.currentTarget);
-  };
-  const onUploading = (formData: FormData, fileName: string) => {
-    setUploading(true);
-    axios.post(XSDUploadUrl, formData, {
-      headers: {
-        'Content-Type': 'multipart/form-data',
-      },
-    }).then((response) => {
-      if (response) {
-        props.onXSDUploaded(fileName);
-        setUploadButtonAnchor(null);
-      }
-    }).catch((error) => {
-      if (error) {
-        setErrorText(getLanguageFromKey('form_filler.file_uploader_validation_error_upload', props.language));
-      }
-    }).finally(() => setUploading(false));
-  };
-  const onUploadCancel = () => {
+  }, []);
```

Comment:
I'm not to familiar with useCallback, but how does it work when the second arg array is empty?

Reply:
It means the function will be memoized and not recreated when the component rerenders. This is not specific to useCallback, but also used in useEffect and useMemo.

Usually, any props that can change between renders, and is used in a hook should be in the "dependency array". Here there are no props that are used in the hook, so it can be empty. The "rules of hooks" lint rule will help to call out what dependencies are missing/redundant (https://reactjs.org/docs/hooks-rules.html)